reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt

```diff
@@ -0,0 +1,638 @@
+import fr.xgouchet.elmyr.jvm.useJvmFactories
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+
+/**
+ * Provides the tests for the InternalSdkCore API not related with the writing operations.
+ */
+@RunWith(AndroidJUnit4::class)
+class FeatureSdkCoreContextTest : MockServerTest() {
+
+    @get:Rule
+    var forge = ForgeRule().useJvmFactories().useToolsFactories().withFactory(ConfigurationCoreForgeryFactory())
+
+    private lateinit var stubFeature: Feature
+
+    @StringForgery(type = StringForgeryType.ALPHABETICAL)
+    lateinit var fakeFeatureName: String
+
+    private lateinit var fakeTrackingConsent: TrackingConsent
+
+    private lateinit var testedFeatureSdkCore: FeatureSdkCore
+
+    @Forgery
+    lateinit var fakeConfiguration: Configuration
+
+    @Before
+    fun setUp() {
+        stubFeature = StubStorageBackedFeature(
+            forge,
+            fakeFeatureName,
+            getMockServerWrapper().getServerUrl()
+        )
+        fakeTrackingConsent = forge.aValueFrom(TrackingConsent::class.java)
+        testedFeatureSdkCore = Datadog.initialize(
+            ApplicationProvider.getApplicationContext(),
+            fakeConfiguration,
+            fakeTrackingConsent
+        ) as FeatureSdkCore
+    }
+
+    @After
+    fun tearDown() {
+        Datadog.stopInstance()
+    }
+
+    // region register/get Feature
+
+    @Test
+    fun mustReturnAllTheRegisteredFeatures_when_getFeature_newFeaturesRegistered() {
+        // Given
+        val registeredFeatures = forge.aList(forge.anInt(1, 10)) {
+            StubStorageBackedFeature(
+                forge,
+                forge.anAlphabeticalString(),
+                getMockServerWrapper().getServerUrl()
+            )
+        }.onEach { testedFeatureSdkCore.registerFeature(it) }
+
+        // When
+        val retrievedFeatures = registeredFeatures.mapNotNull { testedFeatureSdkCore.getFeature(it.name)?.unwrap() }
+
+        // Then
+        assertThat(retrievedFeatures).containsAll(registeredFeatures)
+    }
+
+    @Test
+    fun mustReturnTheCrashReportFeature_when_getFeature_crashReportEnabled() {
+        // Given
+        val fakeConfigCrashReportsEnabled = fakeConfiguration.copy(crashReportsEnabled = true)
+
+        // When
+        // stop the current instance
+        Datadog.stopInstance()
+        val internalSdkCore = Datadog.initialize(
+            ApplicationProvider.getApplicationContext(),
+            fakeConfigCrashReportsEnabled,
+            fakeTrackingConsent
+        ) as InternalSdkCore
+
+        // When
+        val crashReportsFeature = internalSdkCore.getFeature(CRASH_REPORTS_TASK_NAME)?.unwrap<Feature>()
+
+        // Then
+        checkNotNull(crashReportsFeature)
```

Comment:
just to be consistent with assertion API usage

```suggestion
       assertThat(crashReportsFeature).isNotNull()
```